presto-parser/src/main/java/io/prestosql/sql/tree/QualifiedName.java

```diff
@@ -56,7 +56,19 @@ public static QualifiedName of(Iterable<Identifier> originalParts)
     private QualifiedName(List<Identifier> originalParts)
     {
         this.originalParts = originalParts;
-        this.parts = originalParts.stream().map(identifier -> identifier.getValue().toLowerCase(ENGLISH)).collect(toImmutableList());
+        if (originalParts.get(0).getValue().toLowerCase(ENGLISH).contains("flex") && originalParts.size() == 3 && originalParts.get(2).getValue().contains("://")) {
+            this.parts = originalParts.stream().map(identifier -> {
+                if (identifier.getValue().contains("://")) {
+                    return identifier.getValue();
+                }
+                else {
+                    return identifier.getValue().toLowerCase(ENGLISH);
+                }
+            }).collect(toImmutableList());
```

Comment:
The engine shouldn't be aware of any specific naming schemes. That should be a connector implementation detail. The underlying issue is that the engine currently doesn't implement proper SQL semantics with regards to delimited (quoted) names. There's a larger effort to fix this here: https://github.com/prestosql/presto/pull/2350